### PR TITLE
[Merged by Bors] - Ben/oracle fix

### PR DESCRIPTION
**Pull request by benluelo**
## Issue

https://app.clickup.com/t/2hhzhg7

### CODE REVIEW

File: frame/oracle/src/tests.rs [hunk 1]
```diff
@@ -1298,14 +1304,13 @@ fn historic_pricing() {
 #[test]
 fn price_of_amount() {
 	new_test_ext().execute_with(|| {
-		let value = 100500;
-		let id = 42;
-		let amount = 10000;
+		// Yes I know this is broken currently
```
Comment (KaiserKarel) [inline frame/oracle/src/tests.rs:1307]: Would prefer a comment describing why it is broken :)

Comment (benluelo) [reply frame/oracle/src/tests.rs:1307]: Part of the issue is I'm not sure why it's broken - I'm going to take another crack at it later and see if I can figure it out

Comment (benluelo) [reply frame/oracle/src/tests.rs]: See https://github.com/ComposableFi/composable/pull/1207/commits/61b671de32eeaef39511196eb244971313ac73a2, still not sure what the issue is

Comment (benluelo) [reply frame/oracle/src/tests.rs]: Added a proptest (https://github.com/ComposableFi/composable/pull/1207/commits/231b0d9e4570c102f34e4ac94eae2d17096633ac), every input fails. I think the issue is https://github.com/ComposableFi/composable/blob/main/frame/oracle/src/lib.rs#L1010-L1018 ? I'm not sure what this function is for/ what it's expected inputs & outputs are; it's only used [here](https://github.com/ComposableFi/composable/blob/982c67b19051461d8583d46088c7d43679f22082/frame/oracle/src/lib.rs#L419) and [here](https://github.com/ComposableFi/composable/blob/982c67b19051461d8583d46088c7d43679f22082/frame/oracle/src/lib.rs)

File: frame/oracle/src/lib.rs [hunk 2]
```diff
@@ -441,19 +442,25 @@ pub mod pallet {
...
+			// ACENT) is then: 4_000_000_000_000/1_000 == 4_000_000_000; or 4/1000ths (1/250th) of 1
+			// unit of normalized asset.
+			//
+			// if we want 10 units of normalized asset:
+			// 10_000_000_000_000 * (1_000 / 4_000_000_000_000) = 2_500
+			// 10^12 * (10^3 / 4^12) = 2_500
+			//
+			// 2_500 ACENTS are needed to pay for 10 normalized asset.
+			let unit = T::LocalAssets::unit(asset_id)?;
+			let price_asset_for_unit: u128 = Self::get_price(asset_id, unit)?.price.into();
```
Comment (vimukthi-git) [inline frame/oracle/src/lib.rs:459]:
```suggestion
			let asset_price_per_unit: u128 = Self::get_price(asset_id, unit)?.price.into();
```
?

File: frame/oracle/src/lib.rs [hunk 3]
```diff
@@ -441,19 +442,25 @@ pub mod pallet {
...
-			// so 1 costs 0,
-			// and amount of normalized desired is 10
-			// 10 * 1_000 / 4 = 2_500
-			// so we need 2_500 asset amount to pay for 10 normalized
-			let unit = 10 ^ (T::LocalAssets::decimals(asset_id))?;
-			let price_asset_for_unit: u128 = Self::get_price(asset_id, unit.into())?.price.into();
+			// assume the following parameters:
+			//
+			// asset A has 3 decimal places
+			// 1 unit of asset ACOIN costs 4 units of normalized asset (4_000_000_000_000)
```
Comment (vimukthi-git) [inline frame/oracle/src/lib.rs:448]: Not sure what's meant by "normalized" here.
```suggestion
			// 1 unit of asset ACOIN costs 4 units of default quote asset (4_000_000_000_000)
```
?

File: frame/oracle/src/lib.rs [hunk 4]
```diff
@@ -441,19 +442,25 @@ pub mod pallet {
...
+			let unit = T::LocalAssets::unit(asset_id)?;
+			let price_asset_for_unit: u128 = Self::get_price(asset_id, unit)?.price.into();
 
 			let amount: u128 = amount.into();
-			let result = multiply_by_rational(amount, unit as u128, price_asset_for_unit)?;
-			let result: u64 = result
-				.try_into()
-				.map_err(|_| Into::<DispatchError>::into(ArithmeticError::Overflow))?;
+			let result = multiply_by_rational(amount, unit.into(), price_asset_for_unit)?;
+			let result: u64 = result.try_into().map_err(|_| ArithmeticError::Overflow)?;
```
Comment (vimukthi-git) [inline frame/oracle/src/lib.rs:463]: Why is this converting to `u64`?

File: frame/oracle/src/lib.rs (hunk 3 above)
Comment (benluelo) [reply frame/oracle/src/lib.rs:448]: "default quote" is a better way to phrase it indeed, thank you :+1:

**Review by vimukthi-git — APPROVED** [review]
LGTM

**Review by Dom-Roth — APPROVED** [review]

**Pull request closed without merging**